mgmtsystem_nonconformity/models/mgmtsystem_nonconformity_abstract.py

```diff
@@ -38,3 +40,50 @@ def action_view_non_conformities(self):
         action["domain"] = self._get_non_conformities_domain()
         action["context"] = self._get_non_conformities_context()
         return action
+
+
+class MailThread(models.AbstractModel):
+    _name = "mail.thread"
+    _inherit = ["mail.thread", "mgmtsystem.nonconformity.abstract"]
+
+    @api.model
+    def get_view(self, view_id=None, view_type="form", **options):
+        res = super().get_view(view_id=view_id, view_type=view_type, **options)
+        if view_type == "form" and self.env.user.has_group(
+            "mgmtsystem.group_mgmtsystem_viewer"
+        ):
+            View = self.env["ir.ui.view"]
+            if view_id and res.get("base_model", self._name) != self._name:
+                View = View.with_context(base_model_name=res["base_model"])
+            doc = etree.XML(res["arch"])
+            all_models = res["models"].copy()
+            for node in doc.xpath("/form/div[hasclass('oe_chatter')]"):
+                # _add_tier_validation_label process
+                new_node = etree.fromstring(
+                    "<field name='non_conformity_count' invisible='1'/>"
+                )
+                new_arch, new_models = View.postprocess_and_fields(new_node, self._name)
+                new_node = etree.fromstring(new_arch)
+                for model in new_models:
+                    if model in all_models:
```

Comment:
Wouldn't it be better to perform a filter/ed in for and so you don't pivot on this condition ´if model in all_models:´?